core/ppu.hpp

```diff
@@ -22,17 +22,103 @@ class PPU {
   // have to update the buffer when SDL asks for it
   u8* getFrameBuffer();
 
-  // probably a lot of other public interface stuff that you'll discover in implementation
-  // See https://gbdev.io/pandocs/Rendering.html
+  // TODO: decide whether this logic should live in the PPU or if we should have the MMU handle 
+  //       these by allowing it to access the current mode of the PPU
+  // cpuReadVRAM()
+  // cpuWriteVRAM()
+  // cpuReadOAM()
+  // cpuWriteOAM()  
     
 private:
   CPU cpu;
   MMU mmu;
   Mode mode;
 
+  unsigned int cyclesLeft;
+
+  const int LCD_WIDTH = 144;
+  const int LCD_HEIGHT = 160;
+
+  const int OAM_CLOCKS = 80;
+  const int VRAM_CLCOKS = 172;
+  const int HBLANK_CLOCKS = 204;
+  const int VBLANK_CLOCKS = 4560;
+
+  const u16 OAM_TABLE = 0xFE00;
+
+  // LCDC (0xFF40) - LCD Control
+    // Bit 7	LCD and PPU enable	0=Off, 1=On
+    // Bit 6	Window tile map area	0=9800-9BFF, 1=9C00-9FFF
+    // Bit 5	Window enable	0=Off, 1=On
+    // Bit 4	BG and Window tile data area	0=8800-97FF, 1=8000-8FFF
+    // Bit 3	BG tile map area	0=9800-9BFF, 1=9C00-9FFF
+    // Bit 2	OBJ size	0=8x8, 1=8x16
+    // Bit 1	OBJ enable	0=Off, 1=On
+    // Bit 0	BG and Window enable/priority	0=Off, 1=On
+  u16 lcdc = mmu.read(0xFF40);
+
+  // STAT (0xFF41) - LCD Status
+    // Bit 6 - LYC=LY STAT Interrupt source         (1=Enable) (Read/Write)
+    // Bit 5 - Mode 2 OAM STAT Interrupt source     (1=Enable) (Read/Write)
+    // Bit 4 - Mode 1 VBlank STAT Interrupt source  (1=Enable) (Read/Write)
+    // Bit 3 - Mode 0 HBlank STAT Interrupt source  (1=Enable) (Read/Write)
+    // Bit 2 - LYC=LY Flag                          (0=Different, 1=Equal) (Read Only)
+    // Bit 1-0 - Mode Flag                          (Mode 0-3, see below) (Read Only)
+    //           0: HBlank
+    //           1: VBlank
+    //           2: Searching OAM
+    //           3: Transferring Data to LCD Controller
+  u16 stat = mmu.read(0xFF41);
```

Comment:
You're definitely the expert here, so I'll try to say what I think I understand, but I hope you correct me where I'm missing something:

The STAT "register" (that's really just a memory location, despite the PanDocs calling it a register) can optionally pause rendering, setting an interrupt for when it should resume? Is that why the PPU needs a way to set interrupts? If that's the case, I think we could delegate that to the CPU, since it's in charge of checking interrupts and has a reference to the MMU (to read STAT) and the PPU (to set the mode).